src/runtime/virtcontainers/sandbox.go

```diff
@@ -589,6 +590,34 @@ func (s *Sandbox) createCgroupManager() error {
 		if spec.Linux.Resources != nil {
 			resources.Devices = spec.Linux.Resources.Devices
 
+			// spec.Linux.Resources.Devices default only contain {"devices":[{"allow":false,"access":"rwm"}]}
+			if len(resources.Devices) == 1 {
```

Comment:
Do we need to make this check here? Looks like by default all devices are denied access with `allow":false`. If we understand from your investigation that virtiofsd needs access to `/dev/null` and `/dev/urandom`, lets just add those by default here. Is may happen, that in some cases, the container is given access to certain devices on the host  in which case your check here will fail. I suggest adding access to devices required by virtiosfd without checking for this condition here.

Reply:
@amshinde, please, would be possible to open an issue for this and then we can address this following the `main` -> `stable-2.2` flow?

The `main` counterpart of this PR has been merged already and having the backport differing from what was merged on `main` may not be optimal.

Reply:
fwiw, with PR #2332, we're adding the following devices to the sandbox cgroup resources:

```
defaultDevices := []string{
		"/dev/null",
		"/dev/random",
		"/dev/full",
		"/dev/tty",
		"/dev/zero",
		"/dev/urandom",
		"/dev/console",
	}
```

Reply:
Given that the advice of @amshinde , I can do some fix to judge the target device is exist or not in cgroup in a new PR.

Reply:
@amshinde how about fix pr https://github.com/kata-containers/kata-containers/pull/2616